src/ai/backend/manager/api/vfolder.py

```diff
@@ -316,6 +323,49 @@ async def _wrapped(request: web.Request, *args: P.args, **kwargs: P.kwargs) -> w
     return _wrapped
 
 
+@web.middleware
+async def audit_log_middleware(request: web.Request, handler: Handler) -> web.StreamResponse:
+    # TODO: in near future we can change this condition to dict so we can add more conditions
+
+    if request.method == "GET":
+        return await handler(request)
+
+    root_ctx: RootContext = request.app["_root.context"]
+    user_uuid = str(request["user"]["uuid"])
+    access_key = request["keypair"]["access_key"]
+    user_email = request["user"]["email"]
+
+    audit_log_data.set({
+        "user_id": user_uuid,
+        "access_key": access_key,
+        "email": user_email,
+        "action": None,
+        "data": {"before": {}, "after": {}},
+        "target_type": None,
+        "target": None,
+        "created_at": datetime.utcnow(),
+        "success": False,
+        "rest_api_path": request.path,
+    })
+
+    try:
+        return await handler(request)
+    except Exception:
+        raise
+    finally:
```

Comment:
```suggestion
    try:
        return await handler(request)
    finally:
```

you do not need to catch exception here unless you handle it